File: Js/calculator.js

function calc(num1, num2, op) {
    switch(op) {
        case '+': return num1 + num2;
        case '-': return num1 - num2;
        case '*': return num1 * num2;
        case '/':
            if (num2 !== 0) {
                return num1 / num2;
            } else {
                return "ZeroDivisionError";
            }
        default:
            return "Operation Error";
    }
}
let num1 = parseFloat(prompt("Write a:"));
let num2 = parseFloat(prompt("Write b:"));
let op = prompt("Write op: (+-*/):");

let answer = calc(num1, num2, op);
alert(`Answer: ${answer}`);

File: Golang/calculator.go
package main

import (
    "fmt"
)

func main() {
    var num1, num2 float64
    var operator string

    fmt.Print("Write a: ")
    fmt.Scanln(&num1)

    fmt.Print("Write b: ")
    fmt.Scanln(&num2)

	fmt.Print("Write operation (-+*/): ")
    fmt.Scanln(&operator)

    answer := calc(num1, num2, operator)
    fmt.Printf("Answer: %v", answer)
}

func calc(num1, num2 float64, operator string) float64 {
    switch operator {
    case "+":
        return num1 + num2
    case "-":
        return num1 - num2
    case "*":
        return num1 * num2
    case "/":
        if num2 != 0 {
            return num1 / num2
        } else {
            fmt.Println("ZeroDivisionError")
            return 0
        }
    default:
        fmt.Println("Operation Error")
        return 0
    }
}